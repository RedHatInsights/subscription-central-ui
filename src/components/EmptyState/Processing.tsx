import React, { FunctionComponent } from 'react';
import { Title, EmptyState, EmptyStateIcon } from '@patternfly/react-core';

const Processing: FunctionComponent = () => {
  const Spinner = () => (
    <span className="pf-c-spinner" role="progressbar" aria-valuetext="Loading...">
      <span className="pf-c-spinner__clipper" />
      <span className="pf-c-spinner__lead-ball" />
      <span className="pf-c-spinner__tail-ball" />
    </span>
  );

  return (
    <EmptyState>
      <EmptyStateIcon variant="container" component={Spinner} />
      <Title size="lg" headingLevel="h4">
        Loading
      </Title>
    </EmptyState>
  );
};

export default Processing;
