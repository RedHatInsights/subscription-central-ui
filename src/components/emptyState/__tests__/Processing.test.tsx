import React from 'react';
import { render } from '@testing-library/react';
import Processing from '../Processing';

it('renders correctly', () => {
  const { container } = render(<Processing />);
  expect(container).toMatchSnapshot();
});
