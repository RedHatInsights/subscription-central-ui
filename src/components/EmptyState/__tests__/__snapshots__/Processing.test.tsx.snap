// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="pf-c-empty-state"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <div
        class="pf-c-empty-state__icon"
      >
        <span
          aria-valuetext="Loading..."
          class="pf-c-spinner"
          role="progressbar"
        >
          <span
            class="pf-c-spinner__clipper"
          />
          <span
            class="pf-c-spinner__lead-ball"
          />
          <span
            class="pf-c-spinner__tail-ball"
          />
        </span>
      </div>
      <h4
        class="pf-c-title pf-m-lg"
      >
        Loading
      </h4>
    </div>
  </div>
</div>
`;
